File: leetcode_submissions/gameOfLife.py
from typing import List


class Solution:
    def gameOfLife(self, board: List[List[int]]) -> list:
        """
        Do not return anything, modify board in-place instead.
        """

        import copy

        board_ = copy.deepcopy(board)
        height = len(board)
        width = len(board[0])
        for h in range(height):
            for w in range(width):
                summ = 0
                rows = []
                cols = []
                if h == 0:
                    rows.extend([h, h + 1])
                elif h == height - 1:
                    rows.extend([h - 1, h])
                else:
                    rows.extend([h - 1, h, h + 1])
                if w == 0:
                    cols.extend([w, w + 1])
                elif w == width - 1:
                    cols.extend([w - 1, w])
                else:
                    cols.extend([w - 1, w, w + 1])
                for row in rows:
                    for col in cols:
                        if (row, col) != (h, w):
                            try:
                                summ += board_[row][col]
                            except:
                                pass
                if summ < 2 and board_[h][w] == 1:
                    board[h][w] = 0
                elif summ > 3 and board_[h][w] == 1:
                    board[h][w] = 0
                elif (summ == 2 or summ == 3) and board_[h][w] == 1:
                    board[h][w] = 1
                elif summ == 3 and board_[h][w] == 0:
                    board[h][w] = 1
        return board


s = Solution()
# print(s.gameOfLife([[0, 1, 0], [0, 0, 1], [1, 1, 1], [0, 0, 0]]))

# print(s.gameOfLife([[1, 1], [1, 0]]))

print(
    *s.gameOfLife(
        [
            [0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0],
            [0, 0, 1, 1, 1, 0],
            [0, 1, 1, 1, 0, 0],
            [0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0],
        ]
    ),
    sep="\n"
)
print()
print(
    *s.gameOfLife(
        [
            [0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0],
            [0, 0, 1, 1, 1, 0],
            [0, 1, 1, 1, 0, 0],
            [0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0],
        ]
    ),
    sep="\n"
)
print()
print(
    *[
        [0, 0, 0, 0, 0, 0],
        [0, 0, 0, 1, 0, 0],
        [0, 1, 0, 0, 1, 0],
        [0, 1, 0, 0, 1, 0],
        [0, 0, 1, 0, 0, 0],
        [0, 0, 0, 0, 0, 0],
    ],
    sep="\n"
)

File: leetcode_submissions/gameOfLife.go
package main

import "fmt"

func gameOfLife(board [][]int) [][]int {
	var height = len(board)
	var width = len(board[0])

	board_ := make([][]int, height)
	copy(board_, board_)
	for h := 0; h < height; h++ {
		for w := 0; w < width; w++ {
			summ := 0
			rows := make([]int, 9)
			cols := make([]int, 9)
			if h == 0 {
				rows = append(rows, []int{h, h + 1}...)
			} else if h == height-1 {
				rows = append(rows, []int{h - 1, h}...)
			} else {
				rows = append(rows, []int{h - 1, h, h + 1}...)
			}
			if w == 0 {
				cols = append(cols, []int{w, w + 1}...)
			} else if w == width-1 {
				cols = append(cols, []int{w - 1, w}...)
			} else {
				cols = append(cols, []int{w - 1, w, w + 1}...)
			}
			for row := range rows {
				for col := range cols {
					if row != h || col != w {

					}
				}
			}
		}
	}
}

func main() {
	var arr [][]int = [][]int{
		{0, 1, 0},
		{0, 0, 1},
		{1, 1, 1},
		{0, 0, 0},
	}
	fmt.Println(gameOfLife(arr))
}
